### PR TITLE
DAML REPL support static-time mode

### PR DESCRIPTION
**Pull request by aherrmann-da**
Adds flags `--static-time` and `--wall-clock-time` to `daml repl` to configure the time mode. The default is `--wall-clock-time` as before.

Adds a regression test that tests whether `setTime` works in the REPL, it will only work in static time mode.
The REPL functional tests are structured in a way that does not allow to modify the sandbox flags such as time mode. I've placed the regression test next to the connection tests. If we add further tests depending on static time mode in the future then we may want to extend the functional tests to be more configurable.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/damlc/lib/DA/Cli/Damlc.hs [hunk 1]
```diff
@@ -304,6 +305,19 @@ cmdRepl numProcessors =
                 , clientMetadataPlugin = Nothing
                 }
 
+    timeModeFlag :: Parser ReplClient.ReplTimeMode
+    timeModeFlag =
+        (flag' ReplClient.ReplWallClock $ mconcat
+            [ short 'w'
+            , long "wall-clock-time"
+            , help "Use wall clock time (UTC)."
```
Comment (cocreature) [inline compiler/damlc/lib/DA/Cli/Damlc.hs:313]:
```suggestion
            , help "Use wall clock time (UTC). (this is the default)"
```

File: compiler/damlc/tests/src/DA/Test/Repl.hs [hunk 2]
```diff
@@ -122,3 +128,37 @@ testConnection damlc scriptDar testDar ledgerPort mbTokenFile mbCaCrt = do
...
+    :: FilePath
+    -> FilePath
+    -> FilePath
+    -> Int
+    -> Assertion
+testSetTime damlc scriptDar testDar ledgerPort = do
+    out <- readCreateProcess cp $ unlines
+        [ "import DA.Date"
+        , "import DA.Time"
+        , "setTime (time (date 2000 Feb 2) 0 1 2)"
```
Comment (cocreature) [inline compiler/damlc/tests/src/DA/Test/Repl.hs:149]: Maybe worth calling `getTime` afterwards and asserting on the result?

File: compiler/repl-service/server/src/main/scala/com/digitalasset/daml/lf/ReplServiceMain.scala [hunk 3]
```diff
@@ -36,12 +36,24 @@ object ReplServiceMain extends App {
       accessTokenFile: Option[Path],
       maxInboundMessageSize: Int,
       tlsConfig: Option[TlsConfiguration],
+      // optional so we can detect if both --static-time and --wall-clock-time are passed.
```
Comment (cocreature) [inline compiler/repl-service/server/src/main/scala/com/digitalasset/daml/lf/ReplServiceMain.scala:39]: Not really important here given that this is not a user-facing tool and I’m fairly confident we can write Haskell code that will not pass both but given that you already wrote it we might as well keep it I guess

File: compiler/damlc/tests/src/DA/Test/Repl.hs (hunk 2 above)
Comment (aherrmann-da) [reply compiler/damlc/tests/src/DA/Test/Repl.hs]: Good idea, I've added such a check

**Pull request merged**